### PR TITLE
[ISSUE-62] Add trestle-bot workflows for autosync catalogs and profiles

### PR DESCRIPTION
**Pull request by gvauter**
This PR adds two new GitHub action workflows using the Trestle-Bot autosync command.  One is for syncing catalog content and the second is for syncing profile content.  These actions are executed upon a new pull request into the main branch.

Test runs:
- https://github.com/gvauter/trestle-demo-03/actions/runs/9208861130
- https://github.com/gvauter/trestle-demo-03/actions/runs/9208770264

### CODE REVIEW

**Review by jpower432 — APPROVED** [review]
LGTM

File: .github/workflows/trestlebot-autosync-catalog.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          ref: ${{ github.head_ref }}
+      - name: Run autosync
+        id: autosync
+        uses: RedHatProductSecurity/trestle-bot/actions/autosync@main
+        with:
+          markdown_path: "markdown/catalogs"
+          oscal_model: "catalog"
+          file_pattern: "*.json,markdown/*"
+          branch: ${{ github.head_ref }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jpower432) [inline .github/workflows/trestlebot-autosync-catalog.yml:31]: I missed this on my initial review. You do not need to specific the `github_token` unless there is an API interaction. For git authentication only (i.e. commit pushes) `action/checkout` handles this.

File: .github/workflows/trestlebot-autosync-profile.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          ref: ${{ github.head_ref }}
+      - name: Run autosync
+        id: autosync
+        uses: RedHatProductSecurity/trestle-bot/actions/autosync@main
+        with:
+          markdown_path: "markdown/profiles"
+          oscal_model: "profile"
+          file_pattern: "*.json,markdown/*"
+          branch: ${{ github.head_ref }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jpower432) [inline .github/workflows/trestlebot-autosync-profile.yml:31]: Same

**Review by jpower432 — CHANGES_REQUESTED** [review]
Adding some details I missed in my initial review

File: .github/workflows/trestlebot-autosync-catalog.yml (hunk 1 above)
Comment (gvauter) [reply .github/workflows/trestlebot-autosync-catalog.yml]: Fixed in latest commit.

File: .github/workflows/trestlebot-autosync-profile.yml (hunk 2 above)
Comment (gvauter) [reply .github/workflows/trestlebot-autosync-profile.yml]: Fixed in latest commit.

**Review by jpower432 — APPROVED** [review]
LGTM

**Pull request merged**